pkg/kubelet/cm/memorymanager/memory_manager_test.go

```diff
@@ -213,3 +300,1998 @@ func TestConvertPreReserved(t *testing.T) {
+			nodeAllocatableReservation: v1.ResourceList{},
+			preReservedMemory:          map[int]map[v1.ResourceName]resource.Quantity{},
+			affinity:                   topologymanager.NewFakeManager(),
+			expectedError:              fmt.Errorf("[memorymanager] you should specify the system reserved memory"),
+			expectedReserved:           expectedReserved,
+		},
+		{
+			description:                "Should return an error when policy name is not correct",
+			policyName:                 "fake",
+			machineInfo:                machineInfo,
+			nodeAllocatableReservation: v1.ResourceList{},
+			preReservedMemory:          map[int]map[v1.ResourceName]resource.Quantity{},
+			affinity:                   topologymanager.NewFakeManager(),
+			expectedError:              fmt.Errorf("unknown policy: \"fake\""),
+			expectedReserved:           expectedReserved,
+		},
+		{
+			description:                "Should create manager with \"none\" policy",
+			policyName:                 "none",
+			machineInfo:                machineInfo,
+			nodeAllocatableReservation: v1.ResourceList{},
+			preReservedMemory:          map[int]map[v1.ResourceName]resource.Quantity{},
+			affinity:                   topologymanager.NewFakeManager(),
+			expectedError:              nil,
+			expectedReserved:           expectedReserved,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			stateFileDirectory, err := ioutil.TempDir("/tmp/", "memory_manager_tests")
+			if err != nil {
+				t.Errorf("Cannot create state file: %s", err.Error())
+			}
+			defer os.RemoveAll(stateFileDirectory)
+
+			mgr, err := NewManager(testCase.policyName, &testCase.machineInfo, testCase.nodeAllocatableReservation, testCase.preReservedMemory, stateFileDirectory, testCase.affinity)
+
+			if !reflect.DeepEqual(err, testCase.expectedError) {
+				t.Errorf("Could not create the Memory Manager. Expected error: '%v', but got: '%v'",
+					testCase.expectedError, err)
+			}
+
+			if testCase.expectedError == nil {
+				if mgr != nil {
+					rawMgr := mgr.(*manager)
+					if !reflect.DeepEqual(rawMgr.policy.Name(), testCase.policyName) {
+						t.Errorf("Could not create the Memory Manager. Expected policy name: %v, but got: %v",
+							testCase.policyName, rawMgr.policy.Name())
+					}
+					if testCase.policyName == "static" {
+						if !reflect.DeepEqual(rawMgr.policy.(*staticPolicy).systemReserved, testCase.expectedReserved) {
+							t.Errorf("Could not create the Memory Manager. Expected system reserved: %+v, but got: %+v",
+								testCase.expectedReserved, rawMgr.policy.(*staticPolicy).systemReserved)
+						}
+					}
+				} else {
+					t.Errorf("Could not create the Memory Manager - manager is nil, but it should not be.")
+				}
+
+			}
+		})
+	}
+}
+
+func TestGetTopologyHints(t *testing.T) {
+	testCases := []testMemoryManager{
+		{
+			description: "Successful hint generation",
+			policyName:  "static",
```

Comment:
Can you please use constants under tests as well?

Reply:
Done